src/wallet/rpcwallet.cpp

```diff
@@ -1892,6 +1892,33 @@ static RPCHelpMan keypoolrefill()
 }
 
 
+static RPCHelpMan newkeypool()
+{
+    return RPCHelpMan{"newkeypool",
+                "\nEntirely clears and refills the keypool."+
```

Comment:
We might want to be precise in documenting what "clears" means. I understand the keys will not be dealed out anymore by `getnewaddress`. But what happens to the keys in the old keypool? Are they deleted from the wallet, or still considered for incoming transactions? (I guess the latter)

Reply:
> what happens to the keys in the old keypool? Are they deleted from the wallet, or still considered for incoming transactions? (I guess the latter)

FWIW; if the keys in the pool, not extracted using `getnewaddress`, can be queried using some wallet software other than Core, then concept NACK to make it the default behavior (I think I've read that was going to be the topic of a different future PR).

OTOH, if they can't be queried, are somehow sealed and inaccessible, then why at all hold them (they are non-HD)?

Reply:
@laanwj they're marked as used and remain stored but won't be returned from getnewaddress. I'll try and improve the helptext.

@katesalazar I'm not sure what you mean about other software retrieving them - do you mean another piece of software that opens Core wallet.dat files? Otherwise no, without opening the wallet file there is no way an external program could "guess" what the addresses were going to be? Sorry if I'm misunderstanding your point but I don't see anything wrong.

Reply:
> do you mean another piece of software that opens Core wallet.dat files?

yea

Reply:
@katesalazar Perhaps I'm not understanding you, because I don't see the relevance.

For context, Bitcoin Core wallets distinguish between the set of addresses/keys that are considered part of the wallet (i.e., transactions to them will be treated as incoming payments), and addresses that are given out as new addresses for receiving.

What this PR does is refresh the new addresses that will be given out, but the old ones remain valid for incoming transactions. They can also be queried still by various RPCs that dump things; they just won't be given out by `getnewaddress` and related procedures.

Reply:
OK, I understand Samuel better now. Thanks.